internal/kafka/internal/config/dynamic_scaling_config.go

```diff
@@ -2,47 +2,63 @@ package config
 
 import (
 	"fmt"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/cloudproviders"
 )
 
 type DynamicScalingConfig struct {
-	filePath      string
-	Configuration map[string]InstanceTypeDynamicScalingConfig
+	filePath                                      string
+	ComputeNodesPerInstanceType                   map[string]InstanceTypeDynamicScalingConfig          `yaml:"compute_nodes_per_instance_type"`
+	MachineTypePerCloudProvider                   map[cloudproviders.CloudProviderID]MachineTypeConfig `yaml:"machine_type_per_cloud_provider"`
+	EnableDynamicScaleUpManagerScaleUpTrigger     bool                                                 `yaml:"enable_dynamic_data_plane_scale_up"`
+	EnableDynamicScaleDownManagerScaleDownTrigger bool                                                 `yaml:"enable_dynamic_data_plane_scale_down"`
+	NewDataPlaneOpenShiftVersion                  string                                               `yaml:"new_data_plane_openshift_version"`
 }
 
 func NewDynamicScalingConfig() DynamicScalingConfig {
 	return DynamicScalingConfig{
-		filePath:      "config/dynamic-scaling-configuration.yaml",
-		Configuration: make(map[string]InstanceTypeDynamicScalingConfig),
+		filePath:                                      "config/dynamic-scaling-configuration.yaml",
+		ComputeNodesPerInstanceType:                   map[string]InstanceTypeDynamicScalingConfig{},
+		MachineTypePerCloudProvider:                   map[cloudproviders.CloudProviderID]MachineTypeConfig{},
+		EnableDynamicScaleUpManagerScaleUpTrigger:     true,
+		EnableDynamicScaleDownManagerScaleDownTrigger: true,
+		NewDataPlaneOpenShiftVersion:                  "",
 	}
 }
 
-func (c *DynamicScalingConfig) ForInstanceType(instanceTypeID string) (InstanceTypeDynamicScalingConfig, bool) {
-	if instanceTypeConfig, found := c.Configuration[instanceTypeID]; found {
-		return instanceTypeConfig, true
-	}
-	return InstanceTypeDynamicScalingConfig{}, false
+func (c *DynamicScalingConfig) IsDataplaneScaleUpEnabled() bool {
+	return c.EnableDynamicScaleUpManagerScaleUpTrigger
 }
 
-func (c *DynamicScalingConfig) InstanceTypeConfigs() map[string]InstanceTypeDynamicScalingConfig {
-	newConfig := make(map[string]InstanceTypeDynamicScalingConfig, len(c.Configuration))
-	for k, v := range c.Configuration {
-		newConfig[k] = v
-	}
+func (c *DynamicScalingConfig) IsDataplaneScaleDownEnabled() bool {
```

Comment:
Same comment about specificity of the name

Reply:
Added _Trigger_ specificity in the method name.